crates/astria-sequencer/src/fees/mod.rs

```diff
@@ -0,0 +1,390 @@
+            .await
+            .wrap_err("fee change fees not found, so this action is disabled")?;
+        Ok(())
+    }
+
+    fn variable_component(&self) -> u128 {
+        0
+    }
+}
+
+#[async_trait::async_trait]
+impl FeeHandler for IbcSudoChange {
+    #[instrument(skip_all, err)]
+    async fn check_and_pay_fees<S: StateWrite>(&self, state: S) -> eyre::Result<()> {
+        state
+            .get_ibc_sudo_change_fees()
+            .await
+            .wrap_err("ibc sudo change fees not found, so this action is disabled")?;
+        Ok(())
+    }
+
+    fn variable_component(&self) -> u128 {
+        0
+    }
+}
+
+#[async_trait::async_trait]
+impl FeeHandler for IbcRelayerChange {
+    #[instrument(skip_all, err)]
+    async fn check_and_pay_fees<S: StateWrite>(&self, state: S) -> eyre::Result<()> {
+        state
+            .get_ibc_relayer_change_fees()
+            .await
+            .wrap_err("ibc relayer change fees not found, so this action is disabled")?;
+        Ok(())
+    }
+
+    fn variable_component(&self) -> u128 {
+        0
+    }
+}
+
+#[async_trait::async_trait]
+impl FeeHandler for FeeAssetChange {
+    #[instrument(skip_all, err)]
+    async fn check_and_pay_fees<S: StateWrite>(&self, state: S) -> eyre::Result<()> {
+        state
+            .get_fee_asset_change_fees()
+            .await
+            .wrap_err("fee asset change fees not found, so this action is disabled")?;
+        Ok(())
+    }
+
+    fn variable_component(&self) -> u128 {
+        0
+    }
+}
+
+#[async_trait::async_trait]
+impl FeeHandler for IbcRelay {
+    #[instrument(skip_all, err)]
+    async fn check_and_pay_fees<S: StateWrite>(&self, state: S) -> eyre::Result<()> {
+        state
+            .get_ibc_relay_fees()
+            .await
+            .wrap_err("ibc relay fees not found, so this action is disabled")?;
+        Ok(())
+    }
+
+    fn variable_component(&self) -> u128 {
+        0
+    }
+}
+
+#[instrument(skip_all, err(level = Level::WARN))]
+async fn check_and_pay_fees<S: StateWrite, T: FeeHandler + Protobuf>(
+    act: &T,
+    base: u128,
+    multiplier: u128,
+    mut state: S,
+    fee_asset: &asset::Denom,
+) -> eyre::Result<()> {
+    let total_fees = base
+        .checked_add(act.variable_component().saturating_mul(multiplier))
+        .ok_or_eyre("fee calculation overflow in sum")?;
+    let transaction_context = state
+        .get_transaction_context()
+        .expect("transaction source must be present in state when executing an action");
+    let from = transaction_context.address_bytes();
+    let transaction_id = transaction_context.transaction_id;
+    let source_action_index = transaction_context.source_action_index;
+
+    ensure!(
+        state
+            .get_account_balance(&from, fee_asset)
+            .await
+            .wrap_err("failed to get account balance")?
+            >= total_fees,
+        "insufficient funds for transfer and fee payment",
+    );
```

Comment:
This check is unnecessary and can be removed. `decrease_balance` below returns an error if there are not enough funds.

I suggest to remove this line, and move `state.decrease_balance` under the `state.is_allowed_fee_asset` check.